tests/Functional/FunctionalTestCase.php

```diff
@@ -4,84 +4,71 @@
 
 namespace tests\Libero\ContentApiBundle\Functional;
 
-use DirectoryIterator;
 use InvalidArgumentException;
-use PHPUnit\Framework\TestCase;
-use Symfony\Component\Filesystem\Filesystem;
+use LogicException;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Bundle\FrameworkBundle\Test\TestContainer;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\ResettableContainerInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\HttpKernel\KernelInterface;
 use tests\Libero\ContentApiBundle\Functional\App\Kernel;
 use function tests\Libero\ContentApiBundle\capture_output;
 
-abstract class FunctionalTestCase extends TestCase
+abstract class FunctionalTestCase extends KernelTestCase
 {
-    /** @var Filesystem */
-    private static $filesystem;
+    /** @var ContainerInterface */
+    protected static $container;
 
-    /**
-     * @var KernelInterface[]
-     */
-    private static $kernels = [];
-
-    public static function setUpBeforeClass() : void
-    {
-        self::$filesystem = new Filesystem();
-        parent::setUpBeforeClass();
-        foreach (self::getTestCases() as $testCase) {
-            self::$kernels[$testCase] = self::createKernel(['test_case' => $testCase]);
-        }
-    }
-
-    public static function tearDownAfterClass() : void
-    {
-        parent::tearDownAfterClass();
-
-        foreach (self::$kernels as $kernel) {
-            self::$filesystem->remove($kernel->getCacheDir());
-        }
-
-        self::$kernels = [];
-    }
-
-    final public function getKernel(string $name) : KernelInterface
-    {
-        return self::$kernels[$name];
-    }
-
-    final protected function captureContent(KernelInterface $kernel, Request $request, &$content) : Response
+    final protected function captureContent(Request $request, &$content) : Response
     {
         return capture_output(
-            function () use ($kernel, $request) : Response {
-                return $kernel->handle($request);
+            function () use ($request) : Response {
+                return self::$kernel->handle($request);
             },
             $content
         );
     }
 
-    private static function getTestCases() : iterable
+    final protected static function bootKernel(array $options = []) : KernelInterface
```

Comment:
It doesn't seem to strictly need to be a static method apart from inheriting the same interface as the Symfony-provided version. `static::` can be a way to signal that there is indeed something process-wide going on (storing some state on `self`).

Reply:
A bunch of this is just for compatibility with Symfony < 4.1, will add a comment.